### PR TITLE
[Mono] Only enable Arm intrinsics for full aot mode

### LINKED ISSUE #73003: .NET 7 exception on Android with EnableLLVM=true
### Description

Run with LLVM on an arm64 device, and you can run into:
```
07-22 09:59:55.062  4374  4374 F mono-rt : [ERROR] FATAL UNHANDLED EXCEPTION: System.PlatformNotSupportedException: Arg_PlatformNotSupported
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.Text.ASCIIUtility.GetIndexOfFirstNonAsciiByte_Intrinsified(Byte* , UIntPtr )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.Text.Unicode.Utf8Utility.GetPointerToFirstInvalidByte(Byte* , Int32 , Int32& , Int32& )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.Text.UTF8Encoding.GetCharCount(Byte* , Int32 )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.String.CreateStringFromEncoding(Byte* , Int32 , Encoding )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.Text.Encoding.GetString(Byte* , Int32 )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.String.CreateStringForSByteConstructor(Byte* , Int32 )
07-22 09:59:55.062  4374  4374 F mono-rt :    at System.String.Ctor(SByte* value)
07-22 09:59:55.062  4374  4374 F mono-rt :    at Java.Interop.TypeManager.GetClassName(IntPtr )
07-22 09:59:55.062  4374  4374 F mono-rt :    at Android.Runtime.JNIEnv.RegisterJniNatives(IntPtr , Int32 , IntPtr , IntPtr , Int32 )
```

Somewhere here:

https://github.com/dotnet/runtime/blob/b31a48ae77cf20729387383d15e207f838b0a076/src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs#L88

So I think `AdvSimd.Arm64.IsSupported` is `true`, but because `GetIndexOfFirstNonAsciiByte_Intrinsified()` is not in the AOT profile (it's running under JIT), we get the above exception. It's possible that the above code should also work under JIT.

/cc @fanyang-mono 

### Reproduction Steps

1. `dotnet new android`
2. `dotnet build -t:Run -c Release -p:EnableLLVM=true` to run on an arm64 device (in my case a Pixel 5)

### Expected behavior

You can use Profiled AOT + LLVM together on Android.

### Actual behavior

We get an exception when trying to use Profiled AOT + LLVM together on Android.

### Regression?

I don't hit this same exception in .NET 6, but maybe it's just luck?

### Known Workarounds

You can turn off profiled AOT, such as `-p:AndroidEnableProfiledAot=false` or don't use LLVM.

### Configuration

.NET 7.0.100-rc.1.22374.1

### Other information

_No response_

### PR DESCRIPTION
**Pull request by fanyang-mono**
Fix #73003

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -718,6 +718,14 @@ emit_hardware_intrinsics (
 		if (!info)
 			goto support_probe_complete;
 		id = info->id;
+
+#ifdef TARGET_ARM64
```
Comment (lambdageek) [inline src/mono/mono/mini/simd-intrinsics.c:722]: I'm not sure this is an arm-only problem.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -718,6 +718,14 @@ emit_hardware_intrinsics (
...
+		if (!(cfg->compile_aot && cfg->full_aot)) {
```
Comment (lambdageek) [inline src/mono/mono/mini/simd-intrinsics.c:723]: I think this also needs to check `cfg->interp`:
```suggestion
		if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp)) {
```

Also, not sure if we should enable this for llvmonly, too.  @vargaz what do you think?

**Review by lambdageek — COMMENTED** [review]
I think we need to also check that we're not in `--aot=full,interp` mode.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/mini/simd-intrinsics.c:722]: We'll just do this for arm64 for now.  It's unclear that we can repro the issue on x64 given that some of the intrinsics have JIT support or they're full AOT. (for example maccatalyst-x64 is an obvious candidate)

**Review by lambdageek — APPROVED** [review]

**Comment by SamMonoRT** [comment]
This is ready to be merged.

**Pull request merged**